crawler/rss.py

```diff
@@ -29,26 +31,51 @@ def extract(cls, objs, code, options):
         ITEM_AUTHOR = 3
         
         soup = objs[0]
-        ret = list()      
+        ret = list() 
 
         for entry in soup.findAll(conf.properties[code][ITEM_NAME]):
             article = dict()
             article["category"] = list()
+            article["keywords"] = list()
             for item in entry.children:
-                if item.name == None or item.name == conf.properties[code][ITEM_NAME]:
+                if item.name == None: 
                     continue
+                elif item.name == conf.properties[code][ITEM_PUBLISH]:
+                    article["publish_date"] = cls.parse_date(item.string)
                 elif item.name == "title":
                     article[item.name] = item.string
                 elif item.name == conf.properties[code][ITEM_CONTENT]:
-                    article["content"] = item.get_text().strip()
+                    article["content"] = cls.parse_content(item.get_text().strip())
                 elif item.name == "link":
                     # Use "href" attribute if available
-                    article["link"] = (item["href"] if (item.get("href") != None) else item.string)
+                    article["link"] = (item["href"] 
+                    if (item.get("href") != None) 
+                    else item.string)
                 elif item.name == conf.properties[code][ITEM_AUTHOR]:
                     article["author"] = item.get_text().strip()
                 elif item.name == "category":
-                    article["category"].append(item.string)
+                    article[item.name].append(item.string)
+                elif item.name == "keywords":
+                    article[item.name].append(item.string)
                 else: 
                     continue
             ret.append(article)
-        return ret
\ No newline at end of file
+        return ret
+
+    @classmethod
+    def parse_date(cls, date_string):
+        """
+        Unify date in different format of each news in the
+        same format.
+        """
+        new_date = parser.parse(date_string)
```

Comment:
How do we deal with date format that can't be parsed? I guess all kinds of weird datetime formats are being used per different providers. We need a way to gracefully break when this fails.

Reply:
How about making formatted strings for each newspaper? e.g.` "{Y}-{M}-{D} {H}:{M}:{S} {TZ}"`

Reply:
I'll take a log of the format that can not be parsed.
I'll store such format and make a new parser corresponding to each.